types/cheerio/index.d.ts

```diff
@@ -13,292 +13,295 @@
+        before(content: Document, ...contents: any[]): Cheerio;
+        before(content: Document[], ...contents: any[]): Cheerio;
+        before(content: Cheerio, ...contents: any[]): Cheerio;
+
+        insertBefore(content: string): Cheerio;
+        insertBefore(content: Document): Cheerio;
+        insertBefore(content: Cheerio): Cheerio;
+
+        remove(selector?: string): Cheerio;
+
+        replaceWith(content: string): Cheerio;
+        replaceWith(content: CheerioElement): Cheerio;
+        replaceWith(content: CheerioElement[]): Cheerio;
+        replaceWith(content: Cheerio): Cheerio;
+        replaceWith(content: () => Cheerio): Cheerio;
+
+        empty(): Cheerio;
+
+        html(): string | null;
+        html(html: string): Cheerio;
+
+        text(): string;
+        text(text: string): Cheerio;
+
+        wrap(content: string): Cheerio;
+        wrap(content: Document): Cheerio;
+        wrap(content: Cheerio): Cheerio;
+
+        css(propertyName: string): string;
+        css(propertyNames: string[]): string[];
+        css(propertyName: string, value: string): Cheerio;
+        css(propertyName: string, value: number): Cheerio;
+        css(propertyName: string, func: (index: number, value: string) => string): Cheerio;
+        css(propertyName: string, func: (index: number, value: string) => number): Cheerio;
+        css(properties: Object): Cheerio;
+
+        // Rendering
+
+        // Miscellaneous
+
+        clone(): Cheerio;
+
+        // Not Documented
+
+        toArray(): CheerioElement[];
+    }
+
+    interface CheerioParserOptions {
+        // Document References
+        // Cheerio https://github.com/cheeriojs/cheerio
+        // HTMLParser2 https://github.com/fb55/htmlparser2/wiki/Parser-options
+        // DomHandler https://github.com/fb55/DomHandler
+
+        xmlMode?: boolean;
+        decodeEntities?: boolean;
+        lowerCaseTags?: boolean;
+        lowerCaseAttributeNames?: boolean;
+        recognizeCDATA?: boolean;
+        recognizeSelfClosing?: boolean;
+        normalizeWhitespace?: boolean;
+        withStartIndices?: boolean;
+        withEndIndices?: boolean;
+        ignoreWhitespace?: boolean;
+        _useHtmlParser2?: boolean;
+    }
+
+    interface CheerioSelector {
+        (selector: string): Cheerio;
+        (selector: string, context: string): Cheerio;
+        (selector: string, context: CheerioElement): Cheerio;
+        (selector: string, context: CheerioElement[]): Cheerio;
+        (selector: string, context: Cheerio): Cheerio;
+        (selector: string, context: string, root: string): Cheerio;
+        (selector: string, context: CheerioElement, root: string): Cheerio;
+        (selector: string, context: CheerioElement[], root: string): Cheerio;
+        (selector: string, context: Cheerio, root: string): Cheerio;
+        (selector: any): Cheerio;
+    }
+
+    interface CheerioRoot extends CheerioSelector {
```

Comment:
Could we just call this `Root`? `cheerio.Root` looks a bit nicer than `cheerio.Root` :-)

Reply:
Updated!